docs/IR.md

```diff
@@ -190,7 +190,7 @@ Each main (top-level) graph MUST define the names and types of its inputs and ou
 Name|Type|Description
 |---|---|---|
 name|string|The name of the value/parameter.
-type|Type|The type of the value.
+type|Type|The type of the value **including shape information**.
```

Comment:
This table seems to be the general description of ValueInfo, which currently is directly after this statement
> Each main (top-level) graph MUST define the names and types of its inputs and outputs, which are specified as ‘value info’ structures, having the following properties:

ValueInfo is used for multiple things and not just graph inputs/outputs so it may be better to put a general description of ValueInfo first, and in that say 'Type' may include shape information.

Following that we could have the 'MUST' statement to describe the additional requirements on the main graph inputs and outputs. I'd maybe move Rama's suggestion ("In particular,
the inputs and outputs of a model are required to *have* a shape, indicating the rank of inputs and outputs,
even though the exact dimensions need not be specified.") into that. 

Something like
> Each main (top-level) graph MUST define the names, types and shapes of its inputs and outputs, which are specified as ‘value info’ structures. The main graph inputs and outputs are required to *have* a shape, indicating the rank, even though the exact dimensions need not be specified.

It may be helpful to be more specific about what 'not specified' could equate do. Unfortunately we don't talk about dim_value vs dim_param vs unknown until much later in the document under 'Static tensor shapes'.

Reply:
Being a little pedantic here as it's a spec. Can we group things together slightly better?

Specifically line 188 has the 'MUST' for a graph, then we have the ValueInfo structure (although nothing explicitly says it's the ValueInfo structure), then we have some details about nested subgraphs and other graph requirements. 

Would something like the following be slightly better organized?

'Graphs have the following properties:'
\<table with GraphProto fields>

'ValueInfo has the following properties:'
\<table with ValueInfo fields>

\<Various other requirements>
Each main (top-level) graph MUST define...
Nested subgraphs (specified as attribute values) MUST define...
Each graph MUST specify ...
etc. etc.

Reply:
Good idea. This flow looks clearer to me. Just moved the table.